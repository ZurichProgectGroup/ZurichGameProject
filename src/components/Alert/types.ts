import { HTMLAttributes } from 'react';

export type OwnProps = {
    icon?: string,
    className?: string,
} & HTMLAttributes<HTMLDivElement>;

export type Props = OwnProps;
