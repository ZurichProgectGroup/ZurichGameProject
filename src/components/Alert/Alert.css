.alert {
    --offset: 2px;

    width: 314px;
    height: 200px;
    background: rgba(26, 22, 48);
    box-shadow: 0 0 83px rgba(250, 21, 255, 0.4);
    border-radius: 4px;
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-family: var(--font-family);
    font-size: 36px;
    font-style: normal;
    font-weight: 600;
    line-height: 43px;
    letter-spacing: 0;
    text-align: center;
    text-transform: uppercase;

    &::before {
        --position: calc(0px - var(--offset));

        padding: 10px;
        z-index: -1;
        content: '';
        display: block;
        position: absolute;
        top: var(--position);
        left: var(--position);
        bottom: var(--position);
        right: var(--position);
        border-radius: 4px;
        background: linear-gradient(#FC54FF, #72BDFF);
    }
}
