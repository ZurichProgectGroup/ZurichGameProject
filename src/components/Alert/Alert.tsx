import React from 'react';
import './Alert.css';
import type { Props } from './types';

const Alert = (props: Props) => {
    const {
        icon,
        onClick,
        children='',
        className = 'alert',
        ...alertProps
    } = props;

    return (
        <div {...alertProps} className={className} onClick={onClick}>
          <img src={icon}/>
            {children}
        </div>
    );
};

export default Alert;
