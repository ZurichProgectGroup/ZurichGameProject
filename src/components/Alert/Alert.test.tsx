import React from 'react';
import { render } from '@testing-library/react';
import Alert from './Alert';

it('renders correctly', () => {
    const { asFragment } = render(<Alert>test alert</Alert>);

    expect(asFragment()).toMatchSnapshot();
});
