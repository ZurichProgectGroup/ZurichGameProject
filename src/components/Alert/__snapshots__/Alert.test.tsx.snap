// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="alert"
  >
    <img
      alt=""
    />
    test alert
  </div>
</DocumentFragment>
`;
